avocado/utils/pmem.py

```diff
@@ -0,0 +1,298 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (C) IBM 2020 - Harish <harish@linux.vnet.ibm.com>
+#
+# Author: Harish <harish@linux.vnet.ibm.com>
+#
+
+
+import re
+import json
+import glob
+import logging
+
+from . import process
+from . import genio
+
+log = logging.getLogger('avocado.test')
+
+
+class PMemException(Exception):
+    """
+    Error raised for all PMem failures
+    """
+    def __init__(self, additional_text=None):
+        self.additional_text = additional_text
+
+    def __str__(self):
+        return ("Command failed.\ninfo: %s" % self.additional_text)
+
+
+class PMem:
+    """
+    PMem class which provides function to perform ndctl and daxctl operations
+
+    This class can be used only if ndctl binaries are provided before hand
+    """
+
+    def __init__(self, ndctl='ndctl', daxctl='daxctl'):
```

Comment:
maybe we need to check if `ndctl` and `daxctl` is present in system as we have default value and user has an impression that it always works .if those libraries not their avocado  library will break feeling  we need to handle that as well, either always take parm through user or check if library path not there return  False and don't create Call object

Reply:
Sure, makes sense.